### PR TITLE
Enable Kokkos HIP

### PR DESCRIPTION
**Pull request by streeve**

### CODE REVIEW

File: src/cabanamd_impl.h [hunk 1]
```diff
@@ -66,7 +66,11 @@ void CbnMD<t_System, t_Neighbor>::init( InputCL commandline )
     if ( system->do_print )
     {
         using exe_space = typename t_System::execution_space;
+#ifndef KOKKOS_ENABLE_HIP
         exe_space::print_configuration( std::cout );
+#else
+        std::cout << "Using Kokkos::HIP" << std::endl;
+#endif
```
Comment (dalg24) [inline src/cabanamd_impl.h]: why do you not rely on `Kokkos::Experimental::HIP::print_configuration()`?

Comment (streeve) [reply src/cabanamd_impl.h]: Missed that - thanks

Comment (streeve) [reply src/cabanamd_impl.h]: I thought I had left in a debug print, but I had added this because Kokkos aborts if the `HIP::print_configuration` is called

**Comment by streeve** [comment]
@junghans WIP is stuck here, easy fix?

**Comment by junghans** [comment]
> @junghans WIP is stuck here, easy fix?

Fixed.

File: src/cabanamd_impl.h (hunk 1 above)
Comment (dalg24) [reply src/cabanamd_impl.h]: Oh yeah... https://github.com/kokkos/kokkos/blob/7a1265fd58b503bcf1a0f4a316c50b44fcf089c6/core/src/HIP/Kokkos_HIP_Instance.cpp#L106
:(

Comment (dalg24) [reply src/cabanamd_impl.h]: FYI https://github.com/kokkos/kokkos/pull/3080

**Review by sslattery — APPROVED** [review]

**Pull request merged**